src/amd/compiler/aco_ir.h

```diff
@@ -856,6 +856,76 @@ enum block_kind {
    block_kind_needs_lowering = 1 << 12,
 };
 
+
+struct RegisterDemand {
+   constexpr RegisterDemand() = default;
+   constexpr RegisterDemand(const int16_t v, const int16_t s)
+      : vgpr{v}, sgpr{s} {}
+   int16_t vgpr = 0;
+   int16_t sgpr = 0;
+
+   constexpr friend bool operator==(const RegisterDemand a, const RegisterDemand b) {
+      return a.vgpr == b.vgpr && a.sgpr == b.sgpr;
+   }
+
+   RegisterDemand operator+(const Temp t) {
+      if (t.type() == vgpr)
+         return RegisterDemand( vgpr + t.size(), sgpr );         
+      else
+         return RegisterDemand( vgpr, sgpr + t.size() );
+   }
+
+   RegisterDemand operator+(const RegisterDemand other) {
+      return RegisterDemand(vgpr + other.vgpr, sgpr + other.vgpr);
+   }
+
+   RegisterDemand operator-(const RegisterDemand other) {
+      return RegisterDemand(vgpr - other.vgpr, sgpr - other.vgpr);
+   }
+
+   RegisterDemand& operator+=(const RegisterDemand other) {
+      vgpr += other.vgpr;
+      sgpr += other.vgpr;
+      return *this;
+   }
+
+   RegisterDemand& operator+=(const Temp t) {
+      if (t.type() == vgpr)
+         vgpr += t.size();
+      else
+         sgpr += t.size();
+      return *this;
+   }
+
+   RegisterDemand& operator-=(const RegisterDemand other) {
+      vgpr -= other.vgpr;
+      sgpr -= other.vgpr;
+      return *this;
+   }
+
+   RegisterDemand& operator-=(const Operand op) {
```

Comment:
I don't think we need this function. If an Operand is not a temporary then it doesn't use a register before. Thus, wherever this function is used, we can also use -= op.getTemp()

Reply:
Yeah I was not sure about Constants and whether they can appear at that point, so I played save